doc/reference/storage_zfs.md

```diff
@@ -5,70 +5,111 @@ discourse: 1333
-
-   However note that instance copies use ZFS snapshots too, so you also cannot
-   restore an instance to a snapshot taken before the last copy without having
-   to also delete all its descendants.
-
-   Copying the wanted snapshot into a new instance and then deleting
-   the old instance does however work, at the cost of losing any other
-   snapshot the instance may have had.
-
- - Note that LXD will assume it has full control over the ZFS pool or dataset.
-   It is recommended to not maintain any non-LXD owned filesystem entities in
-   a LXD ZFS pool or dataset since LXD might delete them.
- - When quotas are used on a ZFS dataset LXD will set the ZFS "quota" property.
-   In order to have LXD set the ZFS "refquota" property, either set
-   "zfs.use\_refquota" to "true" for the given dataset or set
-   "volume.zfs.use\_refquota" to true on the storage pool. The former option
-   will make LXD use refquota only for the given storage volume the latter will
-   make LXD use refquota for all storage volumes in the storage pool. Also you can
-   set "zfs.reserve\_space" on the volume or "volume.zfs.reserve\_space" on the
-   storage pool to use ZFS "reservation"/"refreservation" along with
-   "quota"/"refquota".
- - I/O quotas (IOps/MBs) are unlikely to affect ZFS filesystems very
-   much. That's because of ZFS being a port of a Solaris module (using SPL)
-   and not a native Linux filesystem using the Linux VFS API which is where
-   I/O limits are applied.
-
-## Storage pool configuration
+{abbr}`ZFS (Zettabyte file system)` combines both physical volume management and a file system.
+A ZFS installation can span across a series of storage devices and is very scalable, allowing you to add disks to expand the available space in the storage pool immediately.
+
+ZFS is a block-based file system that protects against data corruption by using checksums to verify, confirm and correct every operation.
+To run at a sufficient speed, this mechanism requires a powerful environment with a lot of RAM.
+
+In addition, ZFS offers snapshots and replication, RAID management, copy-on-write clones, compression and other features.
+
+## Terminology
+
+ZFS creates logical units based on physical storage devices.
+These logical units are called *ZFS pools* or *zpools*.
+Each zpool is then divided into a number of *datasets*.
+These datasets can be of different types:
+- A *ZFS filesystem* can be seen as a partition or a mounted file system.
+- A *ZFS volume* represents a block device.
+- A *ZFS snapshot* captures a specific state of either a ZFS filesystem or a ZFS volume.
+  ZFS snapshots are read-only.
+- A *ZFS clone* is a writable copy of a ZFS snapshot.
+
+## `zfs` driver in LXD
+
+The `zfs` driver in LXD uses ZFS filesystems for images, and ZFS snapshots and clones to create instances and snapshots.
+By default, LXD enables compression when creating a ZFS pool.
+
+LXD assumes that it has full control over the ZFS pool and dataset.
```

Comment:
Can we add a similar statement to the LVM section, but rather than ZFS pool & dataset, it would be volume group for LVM.